HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -21,6 +21,41 @@ class OAuthAccessTokenMockStore: OAuthAccessTokenStore {
     }
 }
 
+class OAuthAccessTokenInterceptorParser: OAuthAccessTokenParser {
+    
+    private let defaultParser = OAuthAccessTokenDefaultParser()
+    
+    var parseAccessTokenCalled: Bool = false
+    
+    private var shouldIntercept: Bool = false
+    private var interceptToken: OAuthAccessToken?
+    
+    func intercept(parsedToken token: OAuthAccessToken?) {
+        shouldIntercept = true
+        interceptToken = token
+    }
+    
+    func reset() {
+        parseAccessTokenCalled = false
+        shouldIntercept = false
+    }
+    
+    func parse(data: NSData) -> ParsedToken {
```

Comment:
I didn't want apps to depend on Result to implement their own parsers for Heimdallr. But I can change it if you think it's unnecessary.